src/parse_proto_field.ts

```diff
@@ -52,21 +49,31 @@ export function parseProtoField(
       // null and object are treated the same (creating an message).
       // array will be repeated message
       if (obj === null) {
-        const protoType = new ProtoType()
-        const protoMessage = new ProtoMessage()
-        protoMessage.setName(pascalcase(fieldName))
-        protoType.setProtoTypeMessage(protoMessage)
-
-        protoField.setType(protoType)
-        break
+        protoField.setType(createEmptyProtoType(fieldName))
       } else if (Array.isArray(obj)) {
-        // TODO(kkweon): Handle array
+        if (obj.length === 0) {
+          protoField.setType(createEmptyProtoType(fieldName))
+          protoField.setRepeated(true)
+        } else {
+          // use the first element to infer the type
+          // TODO(kkweon): Merge messages when multiple different type appears in the array.
+          const elementProtoField = parseProtoField(obj[0], fieldName, fieldTag)
```

Comment:
nit: just curious if using magic number in code blocks is okay/common. 
e.g. `obj[0]` v.s. `obj[start]`

Reply:
That's why it's "dumb" version. It will be revisited once there is a logic to merge two different types.

Reply:
thanks for the comment. understood 👍